src/functions.php

```diff
@@ -266,8 +266,9 @@ function async(callable $function): callable
  * }
  * ```
  *
- * @param PromiseInterface $promise
- * @return mixed returns whatever the promise resolves to
+ * @template T
+ * @param PromiseInterface<T> $promise
+ * @return T
```

Comment:
👍👍👍

Needed to highlight how much I like this change despite all the nit-picking in this review. Keep it up!